core/src/main/java/org/jdbi/v3/core/collector/OptionalPrimitiveCollectorFactory.java

```diff
@@ -41,16 +43,16 @@
 
     @Override
     public boolean accepts(Type containerType) {
-        return COLLECTORS.containsKey(containerType);
+        return containerType instanceof Class && COLLECTORS.containsKey(containerType);
```

Comment:
I'm curious why PMD cares about this. `Map.get()` takes an `Object`, not `T`.

Reply:
It wasn't so much pmd as it was intellij and my spidey senses. I think it's important to shortcircuit lookups like this. It's not a collection of types but of classes, and not every type is a class, so not every type should be looked up. In this case it makes no practical difference but the general practice is important, to prevent confusing code of which the intent and mechanism isn't totally clear.

Reply:
![image](https://user-images.githubusercontent.com/3950300/41033965-7f2b2298-6988-11e8-973c-3976cfc5f38d.png)

Reply:
I'm not convinced this actually performs better, but it doesn't hurt anything.

Reply:
it wasn't meant for performance anyway :p